Adding steps to configure observability with AWS Security (2.7)

dockerymick opened the pull request:
https://github.com/stolostron/backlog/issues/26197

dockerymick committed:
Adding steps to configure observability with AWS Security

dockerymick committed:
Adding other suggestions

dockerymick committed:
Made changes based on Linting tool

dockerymick committed:
updating anchors

dockerymick committed:
Update known_issues.adoc

updating anchor live

dockerymick committed:
Update whats_new.adoc

live updates

cadawson commented on observability/observability_enable.adoc:
```diff
@@ -189,6 +189,154 @@ stringData:
 +
 For more details, follow the IBM Cloud documentation, https://cloud.ibm.com/objectstorage/create[Cloud Object Storage]. Be sure to use the service credentials to connect with the object storage. For more details, follow the IBM Cloud documentation, https://cloud.ibm.com/objectstorage/create%5BCloud[Cloud Object Store] and https://cloud.ibm.com/docs/cloud-object-storage/iam?topic=cloud-object-stor ...
 
+** For Amazon S3 or S3 compatible, you can also use short term, limited-privilege credentials generated with AWS Security Token Service (AWS STS). Refer to link:https://docs.aws.amazon.com/STS/latest/APIReference/welcome.html[AWS Security Token Service documentation] for more details.
```
For Amazon S3 or S3 compatible ??storage?, you can...

dockerymick replied:
updated

dockerymick commented on observability/observability_enable.adoc:
```diff
@@ -189,6 +189,154 @@ stringData:
+** For Amazon S3 or S3 compatible, you can also use short term, limited-privilege credentials generated with AWS Security Token Service (AWS STS). Refer to link:https://docs.aws.amazon.com/STS/latest/APIReference/welcome.html[AWS Security Token Service documentation] for more details.
++
+Generating access keys using AWS Security Service require additional steps, such as creating a IAM policy that limits access to a S3 bucket, creating a IAM role with a trust policy to generate JWT tokens for {ocp-short} service accounts, and specifying annotations for Observability service accounts that require access to S3 bucket. Below is an example of how observability on
+//Adding view if we used bullet points, will receive feedback in peer review
```
@swopebe would it be best to list the steps as bullet points (197-202) or paragraph form (194)?

cadawson commented on observability/observability_enable.adoc:
```diff
@@ -189,6 +189,154 @@ stringData:
 
+** For Amazon S3 or S3 compatible, you can also use short term, limited-privilege credentials generated with AWS Security Token Service (AWS STS). Refer to link:https://docs.aws.amazon.com/STS/latest/APIReference/welcome.html[AWS Security Token Service documentation] for more details.
++
+Generating access keys using AWS Security Service require additional steps, such as creating a IAM policy that limits access to a S3 bucket, creating a IAM role with a trust policy to generate JWT tokens for {ocp-short} service accounts, and specifying annotations for Observability service accounts that require access to S3 bucket. Below is an example of how observability on
```
...an IAM policy... (x2)
...an S3 bucket...
...an IAM role...
...access to the S3 bucket...?
"The following example" instead of "Below".

dockerymick replied:
Went with the bulleted list

cadawson commented on observability/observability_enable.adoc:
```diff
@@ -189,6 +189,154 @@ stringData:
+Generating access keys using AWS Security Service require additional steps, such as creating a IAM policy that limits access to a S3 bucket, creating a IAM role with a trust policy to generate JWT tokens for {ocp-short} service accounts, and specifying annotations for Observability service accounts that require access to S3 bucket. Below is an example of how observability on
+//Adding view if we used bullet points, will receive feedback in peer review
++
+Generating access keys using AWS Security Service require the following additional steps:
```
requires

dockerymick replied:
updated

cadawson commented on observability/observability_enable.adoc:
```diff
@@ -189,6 +189,154 @@ stringData:
++
+Generating access keys using AWS Security Service require the following additional steps:
+
+- Create an IAM policy that limits access to a S3 bucket
```
an S3 bucket

dockerymick replied:
Updated

cadawson commented on observability/observability_enable.adoc:
```diff
@@ -189,6 +189,154 @@ stringData:
+Generating access keys using AWS Security Service require the following additional steps:
+
+- Create an IAM policy that limits access to a S3 bucket
+- Create an IAM role with a trust policy to generate JWT tokens for {ocp-short} service accounts, and 
```
If you choose this approach, I would remove the ", and".

dockerymick replied:
Updated

cadawson commented on observability/observability_enable.adoc:
```diff
@@ -189,6 +189,154 @@ stringData:
+
+- Create an IAM policy that limits access to a S3 bucket
+- Create an IAM role with a trust policy to generate JWT tokens for {ocp-short} service accounts, and 
+- Specify annotations for the observability service accounts that require access to S3 bucket. You can find an example of how observability on Red Hat OpenShift Service on AWS (ROSA) cluster can be configured to work with AWS STS tokens in the _Set environment_ step. See link:https://www.rosaworkshop.io/[Red Hat OpenShift Service on AWS (ROSA)] for more details, along with link:https://www.rosawo ...
```
..the S3 bucket...?

dockerymick replied:
updated

cadawson commented on observability/observability_enable.adoc:
```diff
@@ -189,6 +189,154 @@ stringData:
+- Create an IAM role with a trust policy to generate JWT tokens for {ocp-short} service accounts, and 
+- Specify annotations for the observability service accounts that require access to S3 bucket. You can find an example of how observability on Red Hat OpenShift Service on AWS (ROSA) cluster can be configured to work with AWS STS tokens in the _Set environment_ step. See link:https://www.rosaworkshop.io/[Red Hat OpenShift Service on AWS (ROSA)] for more details, along with link:https://www.rosawo ...
+
+Complete the following steps to generate access keys using AWS Security Service:
```
...the AWS Security Service:?

dockerymick replied:
updated

cadawson commented on observability/observability_enable.adoc:
```diff
@@ -189,6 +189,154 @@ stringData:
+----
+export POLICY_VERSION=$(date +"%m-%d-%y")
+export TRUST_POLICY_VERSION=$(date +"%m-%d-%y") 
+export CLUSTER_NAME=my-cluster
```
`<my-cluster>` if a variable

cadawson commented on observability/observability_enable.adoc:
```diff
@@ -189,6 +189,154 @@ stringData:
+----
+
+. Create a policy for access to your S3 bucket. Run the following command:
++
```
I think we were trying to avoid the "cat <<EOF..." format for the yaml, for consistency. If you can do that here, it would be better, I think.

dockerymick replied:
thanks for the reminder!

cadawson commented on observability/observability_enable.adoc:
```diff
@@ -189,6 +189,154 @@ stringData:
+. Create a trust policy. Run the following command:
++
+[source,yaml]
+----
```
Same as earlier...cat.

dockerymick replied:
updated

cadawson commented on observability/observability_enable.adoc:
```diff
@@ -249,7 +425,7 @@ All the pods in `open-cluster-management-observability` namespace for Thanos, Gr
 *Note*: If you want to exclude specific managed clusters from collecting the observability data, add the following cluster label to your clusters: `observability: disabled`.
 
-The observability service is enabled. After you enable the observability service the following functionalities are initiated:
+The observability service is enabled. After you enable the observability service, the following functionalities are initiated:
```
following functions

dockerymick replied:
updated

cadawson reviewed (changes requested):
Some comments noted. Thanks!

clyang82 commented:
@subbarao-meduri I remembered we have an user story to address how to configure obs with STS. is my understanding correct? Thanks.

dockerymick committed:
Update observability_enable.adoc

updates after doc peer review

swopebe commented:
Pinged: before you break there is any chance we can move on this or hand it off for someone to follow up?

swopebe committed:
Update observability_enable.adoc

swopebe commented on observability/observability_enable.adoc:
```diff
@@ -247,14 +416,14 @@ All the pods in `open-cluster-management-observability` namespace for Thanos, Gr
 
 . Validate that the observability service is enabled and the data is populated by launching the Grafana dashboards. Click the **Grafana link** that is near the console header, from either the console _Overview_ page or the _Clusters_ page.
 +
```
The source is changing as a result of lots of `*`, which I feel we are overusing across the doc. Gonna see where we can fix it.

dockerymick replied:
updated by brandi

swopebe committed:
Update observability_enable.adoc

swopebe commented on observability/observability_enable.adoc:
```diff
@@ -226,9 +377,27 @@ spec:
+
+See link:../apis/observability.json.adoc#observability-api[Observability API] for more information.
++
+. To deploy on infrastructure machine sets, you must set a label for your set by updating the `nodeSelector` in the `MultiClusterObservability` YAML. Your YAML might resemble the following content:
```
we don't need the italicize `nodeSelector` on line 400, so I removed that

swopebe committed:
Update observability_enable.adoc

<<enabling-observability,Enabling observability>> just changed to highlighting the section.

swopebe committed:
Update observability_enable.adoc

swopebe committed:
Update observability_enable.adoc

swopebe commented:
This  is an example of us not all doing the same thing with modular doc...I think this file could contain it all but in other sections, we are breaking our content to be more modular. It's been a lot of work to keep us all on the same page across teh books. @cadawson would this not be split up?

swopebe reviewed (changes requested):
see conversation--a meeting about this, just 15 mins, may help

swopebe committed:
editing to better format

swopebe commented on observability/observability_enable.adoc:
```diff
@@ -247,14 +415,14 @@ All the pods in `open-cluster-management-observability` namespace for Thanos, Gr
-* All the alert managers from the managed clusters are forwarded to the {product-title-short} hub cluster.
-* All the managed clusters that are connected to the {product-title-short} hub cluster are enabled to send alerts back to the {product-title-short} observability service. You can configure the {product-title-short} Alertmanager to take care of deduplicating, grouping, and routing the alerts to the correct receiver integration such as email, PagerDuty, or OpsGenie. You can also handle silencing an ...
+- All the alert managers from the managed clusters are forwarded to the {product-title-short} hub cluster.
+- All the managed clusters that are connected to the {product-title-short} hub cluster are enabled to send alerts back to the {product-title-short} observability service. You can configure the {product-title-short} Alertmanager to take care of deduplicating, grouping, and routing the alerts to the correct receiver integration such as email, PagerDuty, or OpsGenie. You can also handle silencing an ...
```
This was causing the PR format to change; fixed now, it seems.

dockerymick replied:
Thanks for the fix!

swopebe commented on observability/observability_enable.adoc:
```diff
@@ -265,7 +433,7 @@ See xref:../observability/customize_observability.adoc#forward-alerts[Forwarding
 
-Create your object storage secret named, `thanos-object-storage` in the `open-cluster-management-observability` project. Enter the object storage secret details, then click *Create*. *Note*: See step 4 of the <<enabling-observability,Enabling observability>> section to view an example of a secret.
+Create your object storage secret named, `thanos-object-storage` in the `open-cluster-management-observability` project. Enter the object storage secret details, then click *Create*. *Note*: See step four of the _Enabling observability_ section to view an example of a secret.
 
```
Does this need to be a note? (line 436)

dockerymick replied:
fixed

dockerymick committed:
Update observability_enable.adoc

dockerymick committed:
Update observability_enable.adoc

fixed ordered list alignment

dockerymick committed:
Update observability_enable.adoc

swopebe committed:
format fix

swopebe commented:
let's discuss this in arch call tomorrow.

swopebe committed:
Update observability_enable.adoc

swopebe commented on observability/observability_enable.adoc:
```diff
@@ -188,8 +186,151 @@ stringData:
++
+[source,JSON]
+----
+$SCRATCH_DIR/s3-policy.json
```
Asking about line 230 @subbarao-meduri -- is this needed?

subbarao-meduri replied:
@swopebe @dockerymick - this is just a json file name to create that was referenced in the script later. Perhaps we could instead represent it with  `cat << filename` so customers can directly copy the command while creating the file.

subbarao-meduri replied:
Oops I just noticed your comment above that we are trying to avoid `cat << EOF.. ` format. All we are trying to say is that we must create `s3-policy.json` file with the file content below.

swopebe commented on observability/observability_enable.adoc:
```diff
@@ -188,8 +186,151 @@ stringData:
++
+[source,JSON]
+----
+$SCRATCH_DIR/TrustPolicy.json
```
same here

subbarao-meduri replied:
Same as above, this is just a file name we are trying to create.

swopebe commented:
Still waiting on review, discussed some standards, tech debt open for modular doc fixes.

dockerymick committed:
Update after dev review

dockerymick commented:
@subbarao-meduri thanks for your review and clarification. Will you rereview the PR please?

dockerymick committed:
Update observability_enable.adoc

small update for consistency

dockerymick committed:
Merge branch '2.7_stage' into mj-service-token-26197

dislbenn commented:
Looks good to me. @subbarao-meduri can you provide a review also?

/approve

dockerymick commented:
@dislbenn thanks for your approval

dockerymick commented:
This PR has received a peer review from doc team and approval from dev. Moving forward with merging this PR. I also ran the link check tool
